src/lib/pipeline/m2/material/index.js

```diff
@@ -257,6 +270,58 @@ class M2Material extends THREE.ShaderMaterial {
     }
   }
 
+  registerAnimations(def) {
+    const { uvAnimations, transparencyAnimations, vertexColorAnimation } = def;
+
+    this.registerUVAnimations(uvAnimations);
+    this.registerTransparencyAnimations(transparencyAnimations);
+    this.registerVertexColorAnimation(vertexColorAnimation);
+  }
+
+  registerUVAnimations(uvAnimationIndices) {
+    const { animations, uvAnimationValues } = this.m2;
+
+    uvAnimationIndices.forEach((uvAnimationIndex, opIndex) => {
+      const target = this.uniforms.animatedUVs;
+      const source = uvAnimationValues[uvAnimationIndex];
+
+      animations.on('update', () => {
+        target.value[opIndex] = source.matrix;
+      });
```

Comment:
This is here to apply the next value of an animation right? Is there any reason we need to listen for `update` for every single index?

Another thing I'm wondering about is whether these may prevent the animation manager from being garbage collected as there's no way to deregister the listener.